File: app/src/main/java/com/example/energo_monitoring/model/DeviceTemperatureTransducer.java
package com.example.energo_monitoring.model;

import com.example.energo_monitoring.model.api.DeviceInfo;

public class DeviceTemperatureTransducer extends DeviceInfo {
    private String installationPlace;
    private String values;
    private String measureType;

    public DeviceTemperatureTransducer(String name, int typeId, String installationPlace, String values) {
        super(name, typeId);
        this.installationPlace = installationPlace;
        this.values = values;
    }

    public String getInstallationPlace() {
        return installationPlace;
    }

    public void setInstallationPlace(String installationPlace) {
        this.installationPlace = installationPlace;
    }

    public String getValues() {
        return values;
    }

    public void setValues(String values) {
        this.values = values;
    }

    public String getMeasureType() {
        return measureType;
    }

    public void setMeasureType(String measureType) {
        this.measureType = measureType;
    }

    @Override
    public int getFilledState() {
        if(!(installationPlace == null || installationPlace.isEmpty())
                && !(measureType == null || measureType.isEmpty())
                && !(values == null || values.isEmpty()))
            return 2;
        else if(!(installationPlace == null || installationPlace.isEmpty())
                || !(measureType == null || measureType.isEmpty())
                || !(values == null || values.isEmpty()))
            return 1;
        else
            return 0;
    }
}

File: app/src/main/java/com/example/energo_monitoring/checks/data/devices/DeviceTemperatureTransducer.kt
package com.example.energo_monitoring.checks.data.devices

import androidx.room.Entity
import com.example.energo_monitoring.checks.data.api.DeviceInfo
import androidx.room.PrimaryKey

@Entity
class DeviceTemperatureTransducer : DeviceInfo() {
    @PrimaryKey(autoGenerate = true)
    var id = 0
    @JvmField
    var dataId = 0
    var installationPlace = Field("")
    var values = Field("")
    var length = Field("")
    var comment = Field("")

    //    public int getFilledState() {
    //        if(!(installationPlace == null || installationPlace.isEmpty())
    //                && !(length == null || length.isEmpty())
    //                && !(values == null || values.isEmpty()))
    //            return 2;
    //        else if(!(installationPlace == null || installationPlace.isEmpty())
    //                || !(length == null || length.isEmpty())
    //                || !(values == null || values.isEmpty()))
    //            return 1;
    //        else
    //            return 0;
    //    }
}